contracts/test/RbacModule.test.ts

```diff
@@ -0,0 +1,55 @@
+import { HardhatEthersSigner } from "@nomicfoundation/hardhat-ethers/signers";
+import { expect } from "chai";
+import { ethers } from "hardhat";
+import { RbacModule } from "../typechain-types";
+
+describe("RbacModule", function () {
+  let rbacModule: RbacModule;
+  let owner: HardhatEthersSigner;
+  let delegate: HardhatEthersSigner;
+  let ownerAddress: string;
+  let delegateAddress: string;
+
+  before(async function () {
+    [owner, delegate] = await ethers.getSigners();
+    ownerAddress = await owner.getAddress();
+
+    delegateAddress = await delegate.getAddress();
+
+    rbacModule = (await ethers.deployContract("RbacModule")) as unknown as RbacModule;
+  });
+
+  it("Add delegate", async function () {
+    await rbacModule.addDelegate(delegateAddress);
+
+    const isDelegate = await rbacModule.isDelegate(ownerAddress, delegateAddress);
+    expect(isDelegate).to.be.true;
+  });
+
+  it("Remove delegate", async function () {
+    await rbacModule.addDelegate(delegateAddress);
+
+    await rbacModule.removeDelegate(delegateAddress);
+
+    const isDelegate = await rbacModule.isDelegate(ownerAddress, delegateAddress);
+    expect(isDelegate).to.be.false;
+  });
+
+  it("Delegate can call transfer", async function () {
+    await rbacModule.connect(owner).addDelegate(delegateAddress);
+
+    const mockDestination = "0x0000000000000000000000000000000000000000";
+
+    const amount = ethers.parseEther("1.0");
+
+    await expect(rbacModule.connect(delegate).executeTransfer(ownerAddress, mockDestination, amount)).to.be.revertedWithoutReason();
```

Comment:
when the requirement that the user is a delegate is passed the rbac module makes a call to a safe contract. since this safe contract isnt deployed locally by hardhat this errors out. if we want to fix this we will need to deploy both the safe and the module w hardhat.
should we do that in this pr or a seperate one

Reply:
heres whats happening in the test
1. delegate is added by owner, so owner makes call to rbac to add delegate
2. delegate calls module to make transfer
3. module sees delegate is indeed added, tries execute transaction from safe (owner = safe address here)
4. this fails since owner/safe address is indeed not a safe
i still think this is fine, since in the rbac contract itself everything is working the way we want it to
but the test would be better if a safe added the delegate, then eth was sent from from the safe to the dest address, and then we would query balances of the safe and dest before and after the transaction

deploying a safe just to unit test this function in this way is quite a lot of work and pretty difficult, which is im asking if we should do it in this pr or another one

Reply:
this is fixed now